astropy/convolution/convolve.py

```diff
@@ -159,16 +159,19 @@ def convolve(array, kernel, boundary='fill', fill_value=0.,
 
     # Check that the arguments are lists or Numpy arrays
 
-    if isinstance(array, list):
-        array_internal = np.array(array, dtype=float)
-        array_dtype = array_internal.dtype
-    elif isinstance(array, np.ndarray):
+    if isinstance(array, np.ndarray):
```

Comment:
I think even without following my true duck-typing suggestion, this can be simplified using `np.asanyarray` (which passes through `ndarray` subclasses and does not copy unless needed; it is equivalent to `np.array(..., subok=True, copy=False)`:
```
try:
    array_internal = np.asanyarray(array, dtype=float)
except (TypeError, ValueError):
    raise  TypeError(...)
array_dtype = getattr(array, 'dtype', array_internal.dtype)
```

Reply:
Very good point here!  I'll give it a try and report back here if it has unintended consequences.